### ISSUE TITLE
apply_force_at_point produces unexpected results

### ISSUE THREAD

**Issue by Demindiro** [dimforge/rapier#216]
When using `body.apply_force_at_point(force, position, true)` a body with two forces applied such that there should be no torque still starts to spin wildy. If I copy the code and use my on `world_com` however, it works as expected, i.e.

```rust
let world_com = body_pos.transform_point(&body.mass_properties().local_com);
body.apply_force(force, false);
body.apply_torque((position - world_com).cross(&force), true);
```

This is what I expect & get with my own code:

https://user-images.githubusercontent.com/17316999/125871290-6c32f2a8-c1b1-49f5-8dea-0b87f37ce279.mp4

This is what happens with `apply_force_at_point`:

https://user-images.githubusercontent.com/17316999/125871293-cfb3869a-5884-42c4-a18b-7419eee0b884.mp4

**Comment by DasEtwas** [comment]
I may also have this issue, the torque error in my case is proportional to velocity.

**Comment by Christian7573** [comment]
I've been encountering this issue in 2D as well. Here's the test case I wrote for it: https://github.com/Christian7573/rapier_spin_amp_test. Here's a graph of rotation (blue) and angular velocity (orange) over seconds.
![image](https://user-images.githubusercontent.com/11007603/131755734-3e4e45b3-fdd5-46ae-9f87-70037ce9a0e8.png)

**Comment by sebcrozet** [comment]
Thank you for pointing this out. I think the problem is caused by the fact that we update the mass properties of the rigid-body at the beginning of the timestep, while the position update of the rigid-body is done at the end of the timestep. Therefore, the center-of-mass read by `apply_force_at_point` isn’t correct.

**Comment by DasEtwas** [comment]
For anyone looking for a workaround: Try calling `RigidBody::update_world_mass_properties`. As this method is private, one can use e.g. `.lock_rotations(false, false)` to ensure correct mass properties.

**Comment by sebcrozet** [comment]
Fixed in Rapier v0.12.0 (non-alpha version). Please reopen if the problem persists.
